src/compiler/compiler.rs

```diff
@@ -1261,13 +1297,34 @@ mod test {
         // Windows uses bin, everything else uses lib. Just create both.
         fs::create_dir(f.tempdir.path().join("lib")).unwrap();
         fs::create_dir(f.tempdir.path().join("bin")).unwrap();
-        let rustc = f.mk_bin("rustc").unwrap();
+        let rustc = f.mk_bin("rustc.exe").unwrap();
         let creator = new_creator();
         let runtime = single_threaded_runtime();
         let pool = runtime.handle();
+        populate_rustc_command_mock(&creator, &f);
+        let c = detect_compiler(creator, &rustc, f.tempdir.path(), &[], &[], pool, None)
+            .wait()
+            .unwrap()
+            .0;
+        assert_eq!(CompilerKind::Rust, c.kind());
+    }
+
+    #[test]
+    fn test_is_rustc_like() {
+        assert!(is_rustc_like("rustc"));
+        assert!(is_rustc_like("rustc.exe"));
+        assert!(is_rustc_like("/path/to/rustc.exe"));
+        assert!(is_rustc_like("/path/to/clippy-driver.exe"));
+        assert!(!is_rustc_like("rust"));
```

Comment:
`/path/to/rustc` and `/path/to/clippy-driver` should be covered too, for good measure
as well as the caps or mixed casing version.

Reply:
done

Reply:
done